src/prefect/filesystems.py

```diff
@@ -515,6 +515,9 @@ async def write_path(self, path: str, content: bytes) -> str:
         path = self._resolve_path(path)
         dirpath = path[: path.rindex("/")]
 
+        if self.basepath.startswith("smb://"):
+            dirpath = dirpath.replace(f"smb://{self.settings.get('host', '')}", "")
+
```

Comment:
Can you use `urllib.parse.urlparse` for a little added safety when getting the correct `dirpath`?

```suggestion
        if self.basepath.startswith("smb://"):
            parsed = urllib.parse.urlparse(dirpath)
            dirpath = parsed.path

```